cmd/minikube/cmd/env_test.go

```diff
@@ -0,0 +1,386 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/docker/machine/libmachine"
+	"github.com/docker/machine/libmachine/host"
+	"k8s.io/minikube/pkg/minikube/constants"
+	"k8s.io/minikube/pkg/minikube/tests"
+)
+
+type FakeShellDetector struct {
+	Shell string
+}
+
+func (f FakeShellDetector) GetShell(_ string) (string, error) {
+	return f.Shell, nil
+}
+
+type FakeUsageHinter struct{}
+
+func (FakeUsageHinter) GenerateUsageHint(_ string) string { return "" }
+
+type FakeNoProxyGetter struct {
+	NoProxyVar   string
+	NoProxyValue string
+}
+
+func (f FakeNoProxyGetter) GetNoProxyVar() (string, string) {
+	return f.NoProxyVar, f.NoProxyValue
+}
+
+var defaultAPI = &tests.MockAPI{
+	Hosts: map[string]*host.Host{
+		constants.MachineName: &host.Host{
+			Name:   constants.MachineName,
+			Driver: &tests.MockDriver{},
+		},
+	},
+}
+
+func TestEnvUsageHinter(t *testing.T) {
+	e := &EnvUsageHinter{}
+
+	var tests = []struct {
+		shell    string
+		expected string
+	}{
+		{
+			shell: "", //for default fallthrough
+			expected: `# Run this command to configure your shell:
+# eval $(minikube docker-env)
+`,
+		},
+		{
+			shell: "bash",
+			expected: `# Run this command to configure your shell:
+# eval $(minikube docker-env)
+`,
+		},
+		{
+			shell: "fish",
+			expected: `# Run this command to configure your shell:
+# eval (minikube docker-env)
+`,
+		},
+		{
+			shell: "powershell",
+			expected: `# Run this command to configure your shell:
+# & minikube docker-env | Invoke-Expression
+`,
+		},
+		{
+			shell: "cmd",
+			expected: `REM Run this command to configure your shell:
+REM @FOR /f "tokens=*" %i IN ('minikube docker-env') DO @%i
+`,
+		},
+		{
+			shell: "emacs",
+			expected: `;; Run this command to configure your shell:
```

Comment:
made them constants.  I was hoping to do it in a cleaner way, but the ShellConfig struct probably needs to be reworked if its going to be any cleaner.  I also moved the usage hint to just be constants, because I don't think its worth the complexity to continue to have them configurable